handbook/upgrade/upgrade.md

```diff
@@ -0,0 +1,280 @@
+    `nislmigrate` supports capturing data from individual services as well as all installed services using the `--all` argument flag. For brevity `--all` is used most workflows in this chapter. Depending on your needs you may replace the `--all` argument flag with one or more of the individual service argument flags.
+
+## Recommended Upgrade and Migrations Workflows for your deployment
+
+This chapter recommends workflows for the following upgrade and migration scenarios. You should plan for some downtime of your SystemLink Server. By following these recommendations this downtime can be minimized.
+
+- [Single Node Upgrade](#single-node-upgrade)
+- [Single Node to Multi Node Migration](#single-node-to-multi-node-migration)
+    - [Single Node to Multi Node with MongoDB](#single-node-to-multi-node-with-mongodb)
+    - [Single Node to Multi Node with File Storage](#single-node-to-multi-node-with-file-storage)
+    - [Single Node to Multi Node with PostGreSQL](#single-node-to-multi-node-with-postgresql)
+- [Upgrading Multi node configurations (MongoDB, PostGreSQL, File Share/S3)](#upgrading-multi-node-configurations)
+- [Seamless cut-over](#seamless-cut-over)
+
+!!!note "Storing Captured Data"
+    For upgrades and migrations for a SystemLink Server with a small amount of data you may be able to use `nislmigrate` to backup your data to the local file system of your SystemLink Server. For SystemLink Servers with a large amount of data be aware `nislmigrate` could exhaust the local storage on the server. This can mitigated by using an attached volume. **This document assumes this will be needed and refers to this attached volume as `D:\` in each workflow.**
+
+## Single Node Upgrade
+
+This workflow describes steps to upgrade a single node deployment of SystemLink Server.
+
+The SystemLink NI Package Manager Installers (NIPM) supports in-place upgrades of SystemLink Server. An in place upgrade is one where the upgrade is run directly on your current SystemLink Server. For production deployments of SystemLink Server this is not recommended. If you pursue this option it is highly recommended you backup your server prior to the upgrade. This backup is an essential fallback should an issue occur during upgrade that leaves your SystemLink Server in a bad state.
+
+For single node upgrades, NI recommends pairing the upgrade with a migration. This mitigates risks of issues during the upgrade by ensuring your original SystemLink Server remains in an operable state.
+
+1. Backup your SystemLink Server.
+
+1. Attach a volume to store data captured by `nislmigrate`. This will be referenced as `D:\`.
+
+1. Install `nislmigrate`.
+
+1. Run the command `nislmigrate capture --all --secret <your secret> --dir D:\migration`.
+
+    !!!note
+        While `nislmigrate` runs all SystemLink services are stopped. Plan ahead for this downtime of your SystemLink server. When the capture is completed, SystemLink services will be restarted automatically.
```

Comment:
Should there be a general preparation step for common setup and information like this which is common to all paths?

Reply:
After writing all the workflows down I was wondering the same thing. I'm going to work through the rest of the feedback and then determine if there is a good way to reduce this repetition.